interval/tools.go

```diff
@@ -2,9 +2,28 @@ package interval
 
 import (
 	"fmt"
+	"github.com/vertgenlab/gonomics/numbers"
 )
 
 // CoordsToString takes in a interval and returns a string in the format of chr:start-end
 func CoordsToString(i Interval) string {
 	return fmt.Sprintf("%s:%v-%v", i.GetChrom(), i.GetChromStart(), i.GetChromEnd())
 }
+
+func IntervalSize(i Interval) int {
+	return i.GetChromEnd() - i.GetChromStart()
+}
+
+func OverlapSize(a, b Interval) int {
+	if a.GetChrom() != b.GetChrom() {
+		return 0
+	}
+	end := numbers.Min(a.GetChromEnd(), b.GetChromEnd())
+	start := numbers.Max(a.GetChromStart(), b.GetChromStart())
+	overlapSize := end - start
```

Comment:
You can probably get rid of this and check directly and return the result.
See the suggested changes for _**lines 23, 24 and 27**_
```suggestion
	// overlapSize := end - start
```